File: apps/query-burst/src/main/java/com/booster/queryburst/order/domain/OrderRepository.java
package com.booster.queryburst.order.domain;

import org.springframework.data.domain.Slice;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface OrderRepository extends JpaRepository<Orders, Long> {

    // OFFSET 기반 회원별 주문 목록 (v1 — COUNT 포함)
    Slice<Orders> findSliceByMemberId(Long memberId, Pageable pageable);
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/order/domain/OrderRepository.kt
package com.booster.kotlin.shoppingservice.order.domain

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.time.LocalDateTime

interface OrderRepository : JpaRepository<Order, Long> {
    fun findAllByUserIdOrderByCreatedAtDesc(userId: Long, pageable: Pageable): Page<Order>
    fun findAllByStatusOrderByCreatedAtDesc(status: OrderStatus, pageable: Pageable): Page<Order>
    fun findAllByOrderByCreatedAtDesc(pageable: Pageable): Page<Order>
    fun countByStatus(status: OrderStatus): Long
    fun countByCreatedAtBetween(from: LocalDateTime, to: LocalDateTime): Long
    @Query("SELECT COALESCE(SUM(o.totalPrice - o.discountAmount), 0) FROM Order o WHERE o.status IN :statuses")
    fun sumPaymentAmountByStatusIn(statuses: List<OrderStatus>): Long
}
